zipline/gens/tradesimulation.py

```diff
@@ -208,17 +204,13 @@ def _call_before_trading_start(self, dt):
     def on_dt_changed(self, dt):
         if self.algo.datetime != dt:
             self.algo.on_dt_changed(dt)
+            self.algo._portfolio = None
```

Comment:
seems a bit weird to be mucking with algo's internal state here, and it's not obvious (from here) what is going on.  I think it would be better to have a method on algo that cleared `_portfolio` and `_account`, and call that method from here.